worlds/pokemon_emerald/data.py

```diff
@@ -930,24 +905,36 @@ def _init() -> None:
             data.species[evolution.species_id].pre_evolution = species.species_id
 
     # Replace default item for dex entry locations based on evo stage of species
-    evo_stage_to_ball_map = {
+    evo_stage_to_ball_map: Dict[int, int] = {
         0: data.constants["ITEM_POKE_BALL"],
         1: data.constants["ITEM_GREAT_BALL"],
         2: data.constants["ITEM_ULTRA_BALL"],
     }
+
     for species in data.species.values():
-        evo_stage = 0
+        default_item: Optional[int] = None
         pre_evolution = species.pre_evolution
-        while pre_evolution is not None:
-            evo_stage += 1
-            pre_evolution = data.species[pre_evolution].pre_evolution
+
+        if pre_evolution is not None:
+            evo_data = next(evo for evo in data.species[pre_evolution].evolutions if evo.species_id == species.species_id)
+            if evo_data.method == EvolutionMethodEnum.ITEM:
+                default_item = evo_data.param
+            else:
+                default_item = None
```

Comment:
The default is `None` so can this clause just be removed?

Reply:
Looks like it. Must've missed it while moving stuff around during the refactor.